mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/util/InfrastructureParameterBuilder.java

```diff
@@ -193,6 +202,29 @@ public static ParameterDeclaration addExpirationPolicy(ConfigurationDeclaration
     return parameter;
   }
 
+  public static ParameterDeclaration addErrorMappings(OperationDeclaration operation) {
+    ParameterDeclaration parameter = new ParameterDeclaration(ERROR_MAPPINGS_PARAMETER_NAME);
+    parameter.setDescription(ERROR_MAPPINGS_PARAMETER_DESCRIPTION);
+    parameter.setExpressionSupport(NOT_SUPPORTED);
+    parameter.setRequired(false);
+    parameter.setParameterRole(BEHAVIOUR);
+    parameter.setType(new ErrorMappingsTypeBuilder().buildErrorMappingsType(), false);
```

Comment:
We should cache this type rather than build it every time... maybe pass it as an argument so that we don't have to invalidate caches?

Reply:
will create another issue for this since this applies to all other params build in this class: https://www.mulesoft.org/jira/browse/MULE-18587

Reply:
I don't agree with that approach. This will have a direct impact in memory consumption. I understand that fixing the others is outside the scope of this PR, but don't start by. making it worse here